test/controllers/paye/PayeAtsMainControllerSpec.scala

```diff
@@ -70,5 +82,27 @@ class PayeAtsMainControllerSpec extends UnitSpec with GuiceOneAppPerTest with I1
       document.getElementsByTag("p").get(2).text shouldBe(Messages("paye.ats.index.html.tax_calc_description"))
     }
 
+    "redirect user to noAts page when receiving NOT_FOUND from service" in new TestController {
+
+      when(payeAtsService.getPayeATSData(eqTo(testNino), eqTo(taxYear))(any[HeaderCarrier]))
+        .thenReturn(Left(HttpResponse(responseStatus = NOT_FOUND, responseJson = Some(Json.toJson(NOT_FOUND)))))
+
+      val result = show(fakeAuthenticatedRequest)
+      val document = Jsoup.parse(contentAsString(result))
+
+      status(result) shouldBe SEE_OTHER
+      redirectLocation(result).get shouldBe controllers.paye.routes.PayeErrorController.authorisedNoAts().url
```

Comment:
This will give a better error if redirectLocation is None

```suggestion
      redirectLocation(result) shouldBe Some(controllers.paye.routes.PayeErrorController.authorisedNoAts().url)
```

Reply:
agreed
done